File: easy/1979_find_greatest_common_divisor_of_array/solution.cpp
#include <numeric>
#include <vector>

class Solution {
public:
  int findGCD(std::vector<int> &nums) {
    int minNum = nums[0];
    int maxNum = nums[0];

    for (size_t i = 1; i < nums.size(); ++i) {
      minNum = std::min(minNum, nums[i]);
      maxNum = std::max(maxNum, nums[i]);
    }

    return std::gcd(minNum, maxNum);
  }
};

File: easy/1979_find_greatest_common_divisor_of_array/solution.py
import math
from typing import List


class Solution:
    def findGCD(self, nums: List[int]) -> int:
        minNum = nums[0]
        maxNum = nums[0]

        for i in range(1, len(nums)):
            minNum = min(minNum, nums[i])
            maxNum = max(maxNum, nums[i])

        return math.gcd(minNum, maxNum)
